plugins/urls/__init__.py

```diff
@@ -338,356 +401,190 @@ def _respond_shorten(self, result, source, handler):
+
+            if not self.has_shortener(handler):  # Shouldn't happen
+                caller.respond("Shortener '%s' not found - please set a "
+                               "new one!" % handler)
                 return
 
-            url = self.channels[protocol.name][source.name]["last"]
+            _url = self.channels[protocol.name][source.name]["last"]
 
             if len(args) > 0:
-                url = args[0]
+                _url = args[0]
 
             try:
-                d = self.shorten_url(url, handler)
+                d = self.shorten(_url, handler)
                 d.addCallbacks(self._respond_shorten,
                                self._respond_shorten_fail,
                                callbackArgs=(source, handler),
                                errbackArgs=(source, handler))
             except Exception as e:
-                self.logger.exception(_("Error fetching short URL."))
-                caller.respond(__("Error: %s") % e)
-                return
-
-    def tinyurl(self, url):
-        """
-        Shorten a URL with TinyURL. Don't use this directly.
-        """
-
-        return urllib2.urlopen("http://tinyurl.com/api-create.php?url=" +
-                               urllib.quote_plus(url)).read()
-
-    def parse_title(self, url, use_handler=True):
-        """
-        Get and return the page title for a URL, or the title from a
-        specialized handler, if one is registered.
-
-        This function returns a tuple which may be one of these forms..
-
-        * (title, None) if the title was fetched by a specialized handler
-        * (title, domain) if the title was parsed from the HTML
-        * (None, None) if fetching the title was entirely unsuccessful.
-            This occurs in each of the following cases..
-
-            * When a portscan is detected and stopped
-            * When the page simply has no title
-            * When there is an exception in the chain somewhere
-
-        :param url: The URL to check
-        :param use_handler: Whether to use specialized handlers
+                self.logger.exception("Error fetching short URL.")
+                caller.respond("Error fetching short URL.")
+
+    def check_blacklist(self, _url, context):
+        for entry in context["config"]["blacklist"]:
+            if re.match(entry, _url, flags=str_to_regex_flags("ui")):
+                self.logger.debug(
+                    "Matched blacklist regex: %s" % entry
+                )
+                return True
 
-        :type url: str
-        :type use_handler: bool
+        return False
 
-        :returns: A tuple containing the result
-        :rtype: tuple(None, None), tuple(str, str), tuple(str, None)
-        """
+    @inlineCallbacks
+    def run_handlers(self, _url, context):
+        if self.check_blacklist(_url, context):
+            self.logger.warn(
+                "URL {0} is blacklisted; ignoring...".format(_url)
```

Comment:
This doesn't really need to be a warning. It's not a problem, and it's got nothing to do with the bot or bot owner. Perhaps keeping it as info would be useful in case they go looking for why a title wasn't shown, but I'd be inclined to set it to debug.